src/main/java/eu/xenit/gradle/docker/compose/DockerComposeConventionImpl.java

```diff
@@ -108,23 +124,33 @@ public void fromProject(Project project) {
                 composeTask.doFirst(CONFIGURE_COMPOSE_ACTION_NAME, new Action<Task>() { // No lambda -> see #96
                     @Override
                     public void execute(Task t) {
-                        createSetComposeEnvironmentFromPathAction().execute(dockerBuildImage);
+                        if(environmentVariable != null) {
+                            createSetComposeEnvironmentAction(environmentVariable + "_" + t.getName()).execute(dockerBuildImage);
+                        }else{
+                            createSetComposeEnvironmentFromPathAction().execute(dockerBuildImage);
+                        }
                     }
                 });
             });
         });
 
         // Register shortened environment variables for `buildDockerImage` tasks created with the docker plugin
         pluginClasspathChecker.withPlugin(project, DockerPlugin.class, DockerPlugin.PLUGIN_ID, plugin -> {
-            String environmentName = Util.safeEnvironmentVariableName(project.getPath().substring(1)) + "_DOCKER_IMAGE";
-            fromBuildImage(environmentName, project.getTasks().named("buildDockerImage", DockerBuildImage.class));
+            if(environmentVariable != null) {
+                fromBuildImage(environmentVariable,
+                        project.getTasks().named("buildDockerImage", DockerBuildImage.class));
+            }else{
+                String environmentName = Util.safeEnvironmentVariableName(project.getPath().substring(1)) + "_DOCKER_IMAGE";
+                fromBuildImage(environmentName,
+                        project.getTasks().named("buildDockerImage", DockerBuildImage.class));
+            }
         });
         // Check plugin classpath for docker remote api plugin
         pluginClasspathChecker.checkPlugin(project, DockerRemoteApiPlugin.class, "com.bmuschko.docker-remote-api");
     }
 
     @Override
-    public void fromProject(String projectName) {
-        fromProject(composeSettings.getProject().project(projectName));
+    public void fromProject(String environmentVariable, String projectName) {
```

Comment:
Can you move this to be together with `fromProject(String environmentVariable, Project project)`?